### PR TITLE
[POC] Apply default GSG-based ROS2 format

### PR DESCRIPTION
**Pull request by hidmic**
This pull request reformats `rclcpp` using `ament_clang_format` with the configuration in [this patch](https://github.com/ament/ament_lint/compare/hidmic/clang-format). It attempts to minimize the diff to comply with the linter. 

It remains to be confirmed whether `ament_uncrustify` configuration can be updated to accept these changes.

Connected to https://github.com/ament/ament_lint/issues/146.

### CODE REVIEW

File: rclcpp/include/rclcpp/any_service_callback.hpp [hunk 1]
```diff
@@ -31,51 +31,40 @@ template<typename ServiceT>
...
-      const std::shared_ptr<typename ServiceT::Request>,
-      std::shared_ptr<typename ServiceT::Response>
-    )>;
+  using SharedPtrCallback = std::function<void(
+    const std::shared_ptr<typename ServiceT::Request>,
+    std::shared_ptr<typename ServiceT::Response>)>;
+  using SharedPtrWithRequestHeaderCallback = std::function<void(
+    const std::shared_ptr<rmw_request_id_t>,
+    const std::shared_ptr<typename ServiceT::Request>,
+    std::shared_ptr<typename ServiceT::Response>)>;
```
[the remark this answers is not in the document]
Comment (hidmic) [reply rclcpp/include/rclcpp/any_service_callback.hpp]: Right, I can confirm `clang-format` won't give a developer much freedom of choice.

File: rclcpp/include/rclcpp/any_service_callback.hpp [hunk 2]
```diff
@@ -31,51 +31,40 @@ template<typename ServiceT>
...
     typename CallbackT,
     typename std::enable_if<
-      rclcpp::function_traits::same_arguments<
-        CallbackT,
-        SharedPtrCallback
-      >::value
-    >::type * = nullptr
-  >
+      rclcpp::function_traits::same_arguments<CallbackT, SharedPtrCallback>::value>::type * =
+      nullptr>
```
[the remark this answers is not in the document]
Comment (hidmic) [reply rclcpp/include/rclcpp/any_service_callback.hpp]: I agree on this one too.

File: rclcpp/include/rclcpp/any_service_callback.hpp [hunk 3]
```diff
@@ -31,51 +31,40 @@ template<typename ServiceT>
...
 
   SharedPtrCallback shared_ptr_callback_;
   SharedPtrWithRequestHeaderCallback shared_ptr_with_request_header_callback_;
 
 public:
   AnyServiceCallback()
   : shared_ptr_callback_(nullptr), shared_ptr_with_request_header_callback_(nullptr)
-  {}
+  {
+  }
```
[the remark this answers is not in the document]
Comment (hidmic) [reply rclcpp/include/rclcpp/any_service_callback.hpp]: > then them having the flexibility to make that choice (and defend it if needed, to a reviewer who questions it) is very important in my opinion.

IMHO this is key to this entire discussion. 

It doesn't seem to me like `clang-format` ever intended to provide the grounds for such exchanges to happen. There's a set of rules that code must comply with, period. Authors do not have a saying in this regard, neither do reviewers. And I don't think that's bad on its own, specially as a code base grows large. It keeps our own personal biases out the door e.g. when reviewing you may think X style variation (within permissible boundaries) is fine while I may not, and viceversa.

Now, that may not be ideal nor appropriate for ROS2. If it's not, then maybe `clang-format` is not a good choice. We know  `uncrustify` can be configured to be much more tolerant -- bugs aside.

File: rclcpp/include/rclcpp/callback_group.hpp [hunk 4]
```diff
@@ -43,11 +43,7 @@ class NodeWaitables;
 namespace callback_group
 {
 
-enum class CallbackGroupType
-{
-  MutuallyExclusive,
-  Reentrant
-};
+enum class CallbackGroupType { MutuallyExclusive, Reentrant };
```
[the remark this answers is not in the document]
Comment (jacobperron) [reply rclcpp/include/rclcpp/callback_group.hpp]: Arguably, if types are added/removed the diff would be easier to read in multi-line format.

**Comment by ivanpauno** [comment]
I think we can close this PR, until we have settled if we want to go ahead with the changes in https://github.com/ament/ament_lint/issues/146.

**Pull request closed without merging**